tests/unit/objects/test_projects.py

```diff
@@ -5,9 +5,33 @@
 import pytest
 import responses
 
-from gitlab.v4.objects import Project
+from gitlab.v4.objects import (
+    Project,
+    ProjectFork,
+    ProjectUser,
+    StarredProject,
+    UserProject,
+)
 
 project_content = {"name": "name", "id": 1}
+languages_content = {
+    "python": 80.00,
+    "ruby": 99.99,
+    "CoffeeScript": 0.01,
+}
+user_content = {
+    "name": "first",
+    "id": 1,
+    "state": "active",
+}
+forks_content = [
+    {
+        "id": 1,
+    },
+    {
+        "id": "python-gitlab%2Fpython-gitlab",
```

Comment:
I had a look at the documentation and the description of the id is

> The ID or [URL-encoded path of the project](https://docs.gitlab.com/ee/api/index.html#namespaced-path-encoding).

So I thought it might be good to catch errors if the URL-encoded path of the project makes some problems anywhere. But if the Server don't response with this type of information, maybe its better to delete this part. Or I make a comment on this. I will look at it at the start of the week and I will handle it how my feelings will guide me ;)

Reply:
Yes, as you already guessed, this part of the documentation is for making requests (`If using namespaced API requests...`) rather than receiving responses. The server should consistently respond with an `id` of int type, I think this would be a breaking change on GitLab's side otherwise.

I believe we already test the request behavior in other places because this was an important part of making the CLI work with GET requests.

No rush at all, thanks again for the PR!

Reply:
I implemented it now. There is just one thing I wonder about. Is it just the intelligence of my IDE or doesn't I have to write those small fixtures for `user` and `project`? I added them for clarification but the tests run well also without explicit define them.

Reply:
Looks good now @rnoberger thanks again!

Not sure I fully understand your last question. pytest does a bit of magic to automatically discover fixtures by going up the directory tree looking at each `conftest.py`. This is why IDEs usually don't know what to do with pytest fixtures in test function signatures.

In our case, the `user` and `project` fixtures are already written in https://github.com/python-gitlab/python-gitlab/blob/main/tests/unit/conftest.py#L67-L89 and pytest will automatically use them in tests. Some IDEs might detect that and some won't, depending on plugins installed I assume. :)